client/app/components/account_config_mailboxes.coffee

```diff
@@ -4,150 +4,68 @@ classer = React.addons.classSet
 AccountActionCreator = require '../actions/account_action_creator'
 LayoutActionCreator  = require '../actions/layout_action_creator'
 RouterMixin = require '../mixins/router_mixin'
-MailboxList = require './mailbox_list'
+ShouldComponentUpdate = require '../mixins/should_update_mixin'
 MailboxItem = require './account_config_item'
+MailboxPicker = require './mailbox_picker'
 {SubTitle, Form} = require './basic_components'
 AccountDelete = require './account_config_delete'
+cachedTransform = require '../libs/cached_transform'
 
 
 module.exports = AccountConfigMailboxes = React.createClass
     displayName: 'AccountConfigMailboxes'
 
     mixins: [
-        RouterMixin
-        React.addons.LinkedStateMixin # two-way data binding
+        RouterMixin,
+        React.addons.LinkedStateMixin
+        ShouldComponentUpdate.UnderscoreEqualitySlow
     ]
 
-
-    # Do not update component if nothing has changed.
-    shouldComponentUpdate: (nextProps, nextState) ->
-        isNextState = _.isEqual nextState, @state
-        isNextProps = _.isEqual nextProps, @props
-        return not (isNextState and isNextProps)
+    propTypes:
+        # to change the values on the account itself
+        editedAccount: React.PropTypes.instanceOf(Immutable.Map).isRequired
+        requestChange: React.PropTypes.func.isRequired
+        errors: React.PropTypes.instanceOf(Immutable.Map).isRequired
+        onSubmit: React.PropTypes.func.isRequired
 
     getInitialState: ->
-        @propsToState @props
-
-
-    componentWillReceiveProps: (props) ->
-        @setState @propsToState props
-
-
-    # Turn properties into a react state. It mainly set a n array of
-    # object for each mailbox set in the properties.
-    propsToState: (props) ->
-        state = {}
-        state.mailboxesFlat = {}
-
-        if props.mailboxes.value isnt ''
-
-            props.mailboxes.value.map (mailbox, key) ->
-                id = mailbox.get 'id'
-                state.mailboxesFlat[id] = {}
-                ['id', 'label', 'depth'].map (prop) ->
-                    state.mailboxesFlat[id][prop] = mailbox.get prop
-            .toJS()
-
-        return state
+        newMailboxName: ''
+        newMailboxParent: null
 
+    makeLinkState: (field) ->
+        currentValue = @props.editedAccount.get(field)
+        cachedTransform @, '__cacheLS', currentValue, =>
+            value: currentValue
+            requestChange: (value) =>
+                (changes = {})[field] = value
```

Comment:
@aenario it's hard to read this line. It's the same than `changes = "#{field}": value`?